nat/mapping/port_mapping.go

```diff
@@ -99,6 +105,25 @@ func (p *portMapper) Map(protocol string, port int, name string) (release func()
 	}, true
 }
 
+func (p *portMapper) routerIPPublic() bool {
+	ip, err := p.config.MapInterface.ExternalIP()
+	if err != nil {
+		log.Warn().Err(err).Msg("Couldn't detect router IP address")
+		return false
+	}
+
+	log.Debug().Msgf("Detected router public IP address: %s", ip)
+
+	for _, s := range []string{"10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16"} {
```

Comment:
Maybe better to validate against IP oracle external IP instead of hardcoding these values. Some exotic private networks might be skipped here

Reply:
There are no exotic private networks. Only these are considered by numeration standards. Everything else is either reserved for such things as multicasting or should be considered public.